### DISCUSSION TITLE
Planetary orbital motion and rotation - Potential implementation method / Example

### DISCUSSION THREAD

**Discussion by Starwhip** [tsunamayo/Starship-EVO#5160] [Suggestions]
I'll preface this by saying I don't have a lot of Unity experience. However, I think perhaps the best bet for implementing movement of planet voxels, would be to have them in an entirely separate "level", so that when the level moves, the grid remains the same at the local reference frame. A player who gets close to the planet would be transferred to that planet's local reference frame. (This would probably work with asteroid rings too, if you wanted them to "orbit" planets)

Unity supports multi-scene loading, such as in these links. HOWEVER, it appears that it's not possible to move scenes around at runtime, so maybe that's a dead end.

- https://docs.unity3d.com/Manual/physics-multi-scene.html
- https://docs.unity3d.com/Manual/MultiSceneEditing.html
- https://docs.unity3d.com/ScriptReference/SceneManagement.EditorSceneManager.html

In Godot, there is an example of a voxel solar system with orbital motion: https://github.com/Zylann/solar_system_demo
Godot's level structure involves scenes which you add components to, and then you can add/load those scenes as children of other scenes, so it seems easy to pull off this type of motion in that engine. I don't know if a parallel exists in Unity or if you've already got plans for how to do this.

If it's possible I am sure you'll find the way.

**Comment by tsunamayo** [comment]
@Starwhip Yeah thanks, I use a single scene but my own transform component. The issue will be more physics. Thanks
